src/common/types/types.cpp

```diff
@@ -62,15 +75,19 @@ bool DataType::operator!=(const DataType& other) const {
 DataType& DataType::operator=(DataType&& other) noexcept {
     typeID = other.typeID;
     childType = std::move(other.childType);
+    numElementsInTensor = other.numElementsInTensor;
     return *this;
 }
 
 std::unique_ptr<DataType> DataType::copy() {
-    if (childType) {
-        return make_unique<DataType>(typeID, childType->copy());
-    } else {
-        return std::make_unique<DataType>(typeID);
+    if (childType != nullptr) {
```

Comment:
I would switch on typeID. You might want to apply the change across this file.
Actually let's discuss this with @ray6080

Reply:
I would also prefer switch case here.

Reply:
Changed to switch case here